metadata-editor/conf/routes

```diff
@@ -20,6 +20,12 @@ GET     /metadata/:id/usage-rights                      controllers.EditsControl
 PUT     /metadata/:id/usage-rights                      controllers.EditsController.setUsageRights(id: String)
 DELETE  /metadata/:id/usage-rights                      controllers.EditsController.deleteUsageRights(id: String)
 
+# Collections
+GET     /collections                                    controllers.CollectionsController.getCollections
+POST    /collections                                    controllers.CollectionsController.addCollectionFromJson
+PUT     /collections/*collection                        controllers.CollectionsController.addCollectionFromString(collection: String)
```

Comment:
You're right it's idempotent, but it's slightly odd to `PUT` with an empty body (technically the body could be anything as it's discarded by the code), and have that action result in some magical data materialising at that URI. The RFC says `The PUT method requests that the enclosed entity be stored under the supplied Request-URI.` -- technically the RFC doesn't specify precisely how that enclosed entity may vary from the entity returned by GETing that URI again, but the general assumption (which we have respected elsewhere in the Grid) is to return the exact same entity. The benefit of that is that you can respond with 204 No Content (cheaper) and still be confident of what data is at that resource (e.g. for caching or memoizing).

There's an interesting issue with server-generated information, such as the actionData, which generally led us to prefer POST in those cases.

Anyway this is all a little bit academic and you're removed the endpoint so I'll just shut up now :-)